### PR TITLE
Rearranging the code and renaming certain classes.

### PR DESCRIPTION
**Pull request by kl0u**

### CODE REVIEW

File: src/main/java/com/dataartisans/flink/dataflow/FlinkPipelineExecutionEnvironment.java [hunk 1]
```diff
@@ -19,6 +19,7 @@
 import com.dataartisans.flink.dataflow.translation.FlinkBatchPipelineTranslator;
 import com.dataartisans.flink.dataflow.translation.FlinkStreamingPipelineTranslator;
 import com.google.cloud.dataflow.sdk.Pipeline;
+import com.google.cloud.dataflow.sdk.repackaged.com.google.common.base.Preconditions;
```
Comment (mxm) [inline src/main/java/com/dataartisans/flink/dataflow/FlinkPipelineExecutionEnvironment.java]: This import looks shady :) Probably use com.google.common.base.Preconditions?

File: src/main/java/com/dataartisans/flink/dataflow/FlinkPipelineExecutionEnvironment.java [hunk 2]
```diff
@@ -213,21 +226,25 @@ private void createStreamExecutionEnvironment() {
 		// although we do not use the generated timestamps,
 		// enabling timestamps is needed for the watermarks.
 		this.flinkStreamEnv.getConfig().enableTimestamps();
-
 		this.flinkStreamEnv.setStreamTimeCharacteristic(TimeCharacteristic.EventTime);
 
-		this.flinkStreamEnv.enableCheckpointing(1000);
-		this.flinkStreamEnv.setNumberOfExecutionRetries(5);
+		this.flinkStreamEnv.setNumberOfExecutionRetries(options.getNumberOfExecutionRetries());
+		this.flinkStreamEnv.getConfig().setExecutionRetryDelay(options.getExecutionRetryDelay());
```
Comment (mxm) [inline src/main/java/com/dataartisans/flink/dataflow/FlinkPipelineExecutionEnvironment.java]: Aren't you overwriting Flink defaults here?

Comment (mxm) [inline src/main/java/com/dataartisans/flink/dataflow/FlinkPipelineExecutionEnvironment.java]: Would it make sense to only set if it is specified in the Options?

File: src/main/java/com/dataartisans/flink/dataflow/FlinkPipelineRunner.java [hunk 3]
```diff
@@ -156,6 +156,9 @@ public static FlinkPipelineRunner createForTest(boolean streaming) {
 		// ExecutionEnvironment
 		options.setFlinkMaster("[auto]");
 		options.setStreaming(streaming);
+		options.setCheckpointingInterval(1000L);
+		options.setNumberOfExecutionRetries(5);
+		options.setExecutionRetryDelay(3000L);
```
Comment (mxm) [inline src/main/java/com/dataartisans/flink/dataflow/FlinkPipelineRunner.java]: Do we really want to retry for testing?

File: src/main/java/com/dataartisans/flink/dataflow/translation/FlinkStreamingTranslationContext.java [hunk 4]
```diff
@@ -16,6 +16,7 @@
 package com.dataartisans.flink.dataflow.translation;
 
 import com.google.cloud.dataflow.sdk.options.PipelineOptions;
+import com.google.cloud.dataflow.sdk.repackaged.com.google.common.base.Preconditions;
```
Comment (mxm) [inline src/main/java/com/dataartisans/flink/dataflow/translation/FlinkStreamingTranslationContext.java]: Shaded import

File: src/main/java/com/dataartisans/flink/dataflow/translation/wrappers/streaming/FlinkAbstractParDoWrapper.java [hunk 5]
```diff
@@ -116,10 +116,10 @@ public Instant timestamp() {
 
 		@Override
 		public BoundedWindow window() {
-//			if (!(fn instanceof DoFn.RequiresWindowAccess)) {
-//				throw new UnsupportedOperationException(
-//						"window() is only available in the context of a DoFn marked as RequiresWindow.");
-//			}
+			if (!(fn instanceof DoFn.RequiresWindowAccess)) {
+				throw new UnsupportedOperationException(
+						"window() is only available in the context of a DoFn marked as RequiresWindow.");
```
Comment (mxm) [inline src/main/java/com/dataartisans/flink/dataflow/translation/wrappers/streaming/FlinkAbstractParDoWrapper.java]: RequiresWindowAccess

File: src/main/java/com/dataartisans/flink/dataflow/translation/wrappers/streaming/FlinkAbstractParDoWrapper.java [hunk 6]
```diff
@@ -207,11 +207,11 @@ protected void checkTimestamp(WindowedValue<IN> ref, Instant timestamp) {
 
 		if (windows == null) {
 			try {
-				windows = windowFn.assignWindows(windowFn.new AssignContext() {
+				 windows = windowFn.assignWindows(windowFn.new AssignContext() {
```
Comment (mxm) [inline src/main/java/com/dataartisans/flink/dataflow/translation/wrappers/streaming/FlinkAbstractParDoWrapper.java]: Space indention.

**Comment by mxm** [comment]
Looks good @kl0u. I think we can merge this except for some minor issues.

**Comment by kl0u** [comment]
Thanks @mxm ! I integrated your comments. Please have a look.

File: src/main/java/com/dataartisans/flink/dataflow/translation/wrappers/streaming/FlinkAbstractParDoWrapper.java (hunk 5 above)
Comment (mxm) [inline src/main/java/com/dataartisans/flink/dataflow/translation/wrappers/streaming/FlinkAbstractParDoWrapper.java]: Could you correct the reference in the comment?

**Comment by mxm** [comment]
Looks good to merge.

**Comment by mxm** [comment]
Merged in 25033399fee2453c53c10289cebeb7f01d10b23e.

**Pull request closed without merging**